pkg/distributor/distributor_ingest_storage_test.go

```diff
@@ -0,0 +1,319 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package distributor
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/grafana/dskit/grpcutil"
+	"github.com/grafana/dskit/mtime"
+	"github.com/grafana/dskit/user"
+	"github.com/prometheus/client_golang/prometheus/testutil"
+	"github.com/prometheus/common/model"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/twmb/franz-go/pkg/kerr"
+	"github.com/twmb/franz-go/pkg/kgo"
+	"github.com/twmb/franz-go/pkg/kmsg"
+	"google.golang.org/grpc/codes"
+
+	"github.com/grafana/mimir/pkg/mimirpb"
+	"github.com/grafana/mimir/pkg/util/extract"
+	"github.com/grafana/mimir/pkg/util/testkafka"
+)
+
+// kafkaTopic is the Kafka topic used for ingest storage tests.
+const kafkaTopic = "test"
+
+func TestDistributor_Push_ShouldSupportIngestStorage(t *testing.T) {
+	ctx := user.InjectOrgID(context.Background(), "user")
+
+	// Mock distributor current time (used to get stable metrics assertion).
+	now := time.Now()
+	mtime.NowForce(now)
+	t.Cleanup(func() {
+		mtime.NowReset()
+	})
```

Comment:
```suggestion
	t.Cleanup(mtime.NowReset)
```

Reply:
Done in https://github.com/grafana/mimir/pull/7376/commits/610a72b3b8f2959c75169b2429f3562ae09e9ebd